### PR TITLE
[IA-3999] Convert menu button to typescript

### PR DESCRIPTION
**Pull request by jdcanas**
Quick PR that converts MenuButton to a TS component. This is done as part of the `Analyses.js` migration, as the current JavaScript implementation of MenuButton causes errors when used Typescript files. Also done in an effort to migrate components over time.

I also convert react-utils and `forwardRefWithName` to a version that plays well with typescript.

### CODE REVIEW

**Comment by jdcanas** [comment]
Hmm... something is wrong currently... investigating

edit: resolved, should be good

File: src/libs/react-utils.ts [hunk 1]
```diff
@@ -95,7 +97,8 @@ export const combineRefs = refs => {
   }
 }
 
-export const forwardRefWithName = _.curry((name, WrappedComponent) => {
+// TODO: improve these types further
+export const forwardRefWithName = safeCurry((name: string, WrappedComponent) => {
```
Comment (jdcanas) [inline src/libs/react-utils.ts:101]: I would have liked to type `WrappedComponent` here, but unfortunately its parameterized, which causes backwards-incompatible changes for consumers of `fowardRefWithName` that are already in typescript. Think this is enough of an improvement to move forward with regardless.

Comment (msilva-broad) [reply src/libs/react-utils.ts:101]: hmmm.... would this work:

```
const forwardRefWithNameFn = <T>(name: string, WrappedComponent: T) => T {
  return withDisplayName(name, forwardRef(WrappedComponent)) as T
}
export const forwardRefWithName = safeCurry(forwardRefWithNameFn)

```
it's not perfect in that it doesn't fully type the forwardRef stuff, but at least you are getting type flow on T.  The safeCurry helper will flow generics if they are broken out seperately like this.

File: src/components/MenuButton.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { CSSProperties } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { Clickable, ClickableProps } from 'src/components/common'
+import colors from 'src/libs/colors'
+import { forwardRefWithName } from 'src/libs/react-utils'
+
+
+type MenuButtonProps = ClickableProps<keyof JSX.IntrinsicElements>
+
+const MenuButtonInternal = ({ disabled, children, ...props }: MenuButtonProps, ref) => {
```
Comment (msilva-broad) [inline src/components/MenuButton.ts:10]: Thankfully this component is getting good unit test coverage from other unit tests that are consuming it... but we will want to keep an eye out for components that could use more unit testing as we make these improvements.

File: src/libs/react-utils.ts (hunk 1 above)
Comment (jdcanas) [reply src/libs/react-utils.ts:101]: Hmm... it doesn't seem to work due to the underlying `forwardRef` typing, and any permutation to coerce that inner typing makes the downstream components complain

**Review by aherbst-broad — APPROVED** [review]
I played with this locally, worked ok. I defer to the UIWG folks and @msilva-broad for the code/architectural changes.

**Review by msilva-broad — APPROVED** [review]
incremental improvement is a good step.

**Review by n8landolt — APPROVED** [review]

**Pull request merged**